### PR TITLE
Add TestBigToCompact and TestCompactToBig

### PR DESCRIPTION
**Pull request by someone235**

### CODE REVIEW

File: domain/consensus/utils/math/math_test.go [hunk 1]
```diff
@@ -33,3 +34,46 @@ func TestFastLog2Floor(t *testing.T) {
...
 }
+
+// TestBigToCompact ensures BigToCompact converts big integers to the expected
+// compact representation.
+func TestBigToCompact(t *testing.T) {
+	tests := []struct {
+		in  int64
+		out uint32
+	}{
+		{0, 0},
```
Comment (svarogg) [inline domain/consensus/utils/math/math_test.go:45]: Please add some more vectors in both tests.
Consult with Elichai if you need ideas to edge cases.

**Review by svarogg — CHANGES_REQUESTED** [review]

File: domain/consensus/utils/math/math_test.go (hunk 1 above)
Comment (elichai) [reply domain/consensus/utils/math/math_test.go:45]: There are ~20 tests here:
https://github.com/bitcoin/bitcoin/blob/master/src/test/arith_uint256_tests.cpp#L40

**Review by svarogg — APPROVED** [review]
